craft_application/services/provider.py

```diff
@@ -129,6 +132,9 @@ def instance(
 
         provider.ensure_provider_is_available()
 
+        if clean_existing:
```

Comment:
Future feature request: a different instance name when using the fetch service. Rapid development could then use the default instance, with release versions using the fetch service.